cmd/lotus-shed/indexes.go

```diff
@@ -879,3 +886,375 @@ var backfillTxHashCmd = &cli.Command{
 		return nil
 	},
 }
+
+var pruneTxHashCmd = &cli.Command{
+	Name:  "prune-txhash",
+	Usage: "Prune the txhash.db for txs older than a given height",
+	Flags: []cli.Flag{
+		&cli.IntFlag{
+			Name:  "older-than",
+			Value: 0,
+			Usage: "height to start pruning txhashes older than this epoch, 0 means start from head",
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		srv, err := lcli.GetFullNodeServices(cctx)
+		if err != nil {
+			return err
+		}
+		defer srv.Close() //nolint:errcheck
+
+		api := srv.FullNodeAPI()
+		ctx := lcli.ReqContext(cctx)
+
+		basePath, err := homedir.Expand(cctx.String("repo"))
+		if err != nil {
+			return err
+		}
+
+		startHeight := cctx.Int64("older-than")
+		if startHeight == 0 {
+			currTs, err := api.ChainHead(ctx)
+			if err != nil {
+				return err
+			}
+
+			startHeight += int64(currTs.Height())
+
+			if startHeight == 0 {
+				return xerrors.Errorf("bogus start height %d", startHeight)
+			}
+		}
```

Comment:
how about you refactor this into a utility function since this block is repeated 4 times: `basePath, startHeight, err := parsePruneArgs(cctx)`

Reply:
Done.